### ISSUE TITLE
QM9 dataset hybridization issue, a lot of atoms have HybridizationType.UNSPECIFIED

### ISSUE THREAD

**Issue by Jack-XHP** [rusty1s/pytorch_geometric#2163]
From now on, we recommend using our discussion forum (https://github.com/rusty1s/pytorch_geometric/discussions) for general questions.

## ❓ Questions & Help

When I use the QM9 dataset and try to preprocess the raw data with rdkit, I notice many heavy atoms only get HybridizationType.UNSPECIFIED, which is likely caused by the sanitize=False option used in Chem.SDMolSupplier. After referring to rdkit document, atoms' hybridization type is set during the sanitization process. But enable sanitization will fail for about 1403 molecules in QM9.

**Comment by rusty1s** [comment]
Yeah, that's correct. We cannot really make use of `sanitize=True`, since this will result in a dropping of atoms. If you do not want `hybridization` as a feature because of that, I suggest to filter those features out in a post-processing step, e.g., via:
```python
data.x = torch.cat([data.x[:2], data.x[5:]], dim=-1)
```

**Comment by Jack-XHP** [comment]
I find a way to get the hybridization and aromatic without actual do the sanitize process
```SanitizeStep = Chem.rdmolops.SanitizeFlags.SANITIZE_KEKULIZE | Chem.rdmolops.SanitizeFlags.SANITIZE_SETAROMATICITY | Chem.rdmolops.SanitizeFlags.SANITIZE_SETHYBRIDIZATION```

```Chem.SanitizeMol(mol, SanitizeStep)```

**Comment by rusty1s** [comment]
Cool! Let me know if you are interested in contributing this feature :)
